cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -39,6 +39,12 @@ namespace cudf::io::parquet {
 
 using cudf::io::detail::string_index_pair;
 
+// Max decimal precisions according to the parquet spec:
+// https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#decimal
+auto constexpr MAX_DECIMAL32_PRECISION  = 9;
+auto constexpr MAX_DECIMAL64_PRECISION  = 18;
+auto constexpr MAX_DECIMAL128_PRECISION = 38;  // log10(2^(sizeof(int128_t) * 8 - 1) - 1)
```

Comment:
Parquet writer has these values hardcoded at https://github.com/rapidsai/cudf/blob/branch-23.02/cpp/src/io/parquet/writer_impl.cu#L437
Would be nice to use these constants there as well.